source/helper.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  FileDetailProperty,
+  AsyncFunction,
+} from './types';
+
+import {
+  // Exception as NextcloudError,
+
+  ForbiddenError,
+  NotFoundError,
+} from './errors';
+
+const sanitizePath = encodeURI;
+
+export function createFileDetailProperty(namespace: string, namespaceShort: string, element: string, nativeType?: boolean, defaultValue?: any) : FileDetailProperty {
+  return {
+    namespace,
+    namespaceShort,
+    element,
+    nativeType,
+    default: defaultValue
+  };
+}
+
+export function createOwnCloudFileDetailProperty(element: string, nativeType?: boolean, defaultValue?: any) : FileDetailProperty {
```

Comment:
What about creating a single function that takes owncloud or nextcloud as namespace argument? (OK, that will require at least one wrapping function and one with a default.)
The default should be nextcloud since the library's name is `nextcloud-link` after all ;)

Reply:
The OwnCloud / NextCloud methods are both used for NextCloud (both in the same API call as well). I'm not sure how you envision your request. Don't the three methods already match your criteria? 
If I turn it into one method, the developer would have to supply namespace and shortNamespace. If they have to do that, we would only need `createFileDetailProperty`, the NC and OC versions are just there for ease of usage.

Reply:
Discussed with @kwisatz, this will be left as is, `createFileDetailProperty` serves this purpose.